### PR TITLE
Safe import of LRScheduler

### LINKED ISSUE #29895: Transformers 4.39.0 has breaking change on torch compatibility.
### System Info

torch 1.13 + transformers 4.39.0

### Description

`torch.optim.LRScheduler` was only exposed in torch 2:
- https://github.com/pytorch/pytorch/pull/88503

Previously it was called `_LRScheduler` in torch 1.

This recent PR included in 4.39.0 imports it and lead to a hard requirement on torch 2:
- https://github.com/huggingface/transformers/pull/29588

FYI @younesbelkada 

This issue is captured by our CI where we have dependency on torch 1.

### Expected behavior

Suggest to check on version or API availability before use.
Given it's only a visibility change, a transparent fallback/re-route would be ideal.

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Fixes #29895

cc @ArthurZucker

### CODE REVIEW

**Review by xkszltl — APPROVED** [review]
Should we merge these 2 `is_torch_available` sections?

File: src/transformers/trainer_pt_utils.py [hunk 1]
```diff
@@ -34,21 +34,36 @@
...
+if is_torch_available():
+    from .pytorch_utils import is_torch_greater_or_equal_than_2_0
+
 if is_training_run_on_sagemaker():
     logging.add_handler(StreamHandler(sys.stdout))
 
 if is_torch_xla_available():
     import torch_xla.core.xla_model as xm
 
+if is_torch_available():
```
Comment (ArthurZucker) [inline src/transformers/trainer_pt_utils.py:60]:
```suggestion
if is_torch_available():
    from .pytorch_utils import is_torch_greater_or_equal_than_2_0
```

File: src/transformers/trainer_pt_utils.py [hunk 2]
```diff
@@ -34,21 +34,36 @@
...
+    is_sagemaker_mp_enabled,
+    is_torch_available,
+    is_torch_xla_available,
+    is_training_run_on_sagemaker,
+    logging,
+)
 
 
+if is_torch_available():
+    from .pytorch_utils import is_torch_greater_or_equal_than_2_0
```
Comment (ArthurZucker) [inline src/transformers/trainer_pt_utils.py:52]:
```suggestion
```

**Review by ArthurZucker — APPROVED** [review]
Thanks for the prompt fix

**Pull request merged**